packages/components/src/date-time/time/index.tsx

```diff
@@ -142,9 +145,9 @@ export function TimePicker( {
 				numberValue = from12hTo24h( numberValue, am === 'PM' );
 			}
 
-			const newDate = date.clone()[ method ]( numberValue );
+			const newDate = set( date, { [ method ]: numberValue } );
```

Comment:
`set` will return a new date because it uses `toDate` internally.

https://github.com/date-fns/date-fns/blob/77c098105c9713915dec475c9d7d2fcbeec5ca9b/src/set/index.ts#L47
https://date-fns.org/v2.29.2/docs/toDate

Reply:
Great, thank you for the explanation!